sql-analyzer/jni/SQLValidator.c

```diff
@@ -0,0 +1,323 @@
+/* DO NOT EDIT THIS FILE - it is machine generated */
+#include "com_cubrid_validator_SQLValidator.h"
+
+#include <jni.h>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <windows.h>
+
+#include "dbi.h"
+/* Header for class com_cubrid_validator_SQLValidator */
+
+/* GLOBAL STATE */
+#define DB_CONNECTION_STATUS_NOT_CONNECTED 0
+#define DB_CONNECTION_STATUS_CONNECTED     1
+#define DB_CONNECTION_STATUS_RESET        -1
+
+#define NO_ERROR   0
+#define ER_FAILED -1
+
+#define DEFAULT_LIBRARY_PATH "C:\\CUBRID\\bin\\cubridcs.dll"
+#define DEFAULT_LANG_CHARSET "ko_KR.utf8"
+#define DEFAULT_BUF_LENGTH   1024
+
+enum er_exit_ask {
+	ER_NEVER_EXIT = 0,
+	ER_EXIT_ASK = 1,
+	ER_EXIT_DONT_ASK = 2,
+	ER_ABORT = 3,
+	ER_EXIT_DEFAULT = ER_NEVER_EXIT
+};
+
+typedef struct error_buf {
+	char *buf;
+	int length;
+} ERROR_BUF;
+
+typedef int (*fn_er_init)(const char *msglog_filename, int exit_ask);
+typedef int (*fn_er_errid)();
+typedef const char * (*fn_er_msg)();
+typedef DB_SESSION_ERROR * (*fn_db_get_errors)(DB_SESSION * session);
+typedef DB_SESSION_ERROR * (*fn_db_get_next_error)(DB_SESSION_ERROR * errors,
+		int *line, int *col);
+typedef int (*fn_lang_init)(); /* lang_Initialized */
+typedef int (*fn_lang_set_charset_lang)(const char *lang_charset); /* lang_Language_initialized */
+typedef void (*fn_db_set_connect_status)(int status);
+typedef DB_SESSION * (*fn_db_open_buffer)(char * query);
+typedef void (*fn_db_close_session)(DB_SESSION * session);
+
+static fn_er_init cub_er_init = NULL;
+static fn_er_errid cub_er_errid = NULL;
+static fn_er_msg cub_er_msg = NULL;
+static fn_db_get_errors cub_db_get_errors = NULL;
+static fn_db_get_next_error cub_db_get_next_error = NULL;
+static fn_lang_init cub_lang_init = NULL;
+static fn_lang_set_charset_lang cub_lang_set_charset_lang = NULL;
+static fn_db_set_connect_status cub_db_set_connect_status = NULL;
+static fn_db_open_buffer cub_db_open_buffer = NULL;
+static fn_db_close_session cub_db_close_session = NULL;
+
+static int load(HINSTANCE hDLL, const char * libraryPath);
+static int init(const char *lang_charset);
+static int set_error(char ** error_buf, const char *er_msg);
+static char * jstring_to_char(JNIEnv *env, jstring j_str);
+static jstring char_to_jstring(JNIEnv *env, const char *str);
+
+/*
+ * Class:     com_cubrid_validator_SQLValidator
+ * Method:    validateSQL
+ * Signature: (Ljava/lang/String;)Ljava/lang/String;
+ */
+JNIEXPORT jstring JNICALL Java_com_cubrid_validator_SQLValidator_validateSQL(
+		JNIEnv *env, jobject j_obj, jstring j_str) {
+	HINSTANCE hDLL = NULL;
+	BOOL fFreeResult;
+	char *query = NULL;
+	DB_SESSION *session = NULL;
+	DB_SESSION_ERROR *session_error = NULL;
+	char *out_buf = NULL;
+	int out_buf_len = 0;
+	int out_buf_maxlen = 0;
+	int has_error = 0;
+	jstring result = NULL;
+
+	if (load(hDLL, DEFAULT_LIBRARY_PATH) != NO_ERROR) {
```

Comment:
If users install cubrid in the other directory, they seems to meet error message.
How about changing to use CUBRID environment value instead of DEFAULT_LIBRARY_PATH ?

Reply:
Does users need to start cub_server for using cubridcs.dll ?